packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -234,7 +235,32 @@ id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, s
 
   jsi::Value error = createJSRuntimeError(runtime, "Exception in HostFunction: " + reason);
   error.asObject(runtime).setProperty(runtime, "cause", std::move(cause));
-  return {runtime, std::move(error)};
+  return jsi::JSError(runtime, std::move(error));
+}
+
+/**
+ * Creates JSError with current JS runtime and NSDictionary data as cause.
+ */
+static jsi::JSError convertNSDictionaryToJSError(jsi::Runtime &runtime, NSDictionary *cause)
+{
+  jsi::Value error = createJSRuntimeError(runtime, "Exception in HostFunction: " + (cause[@"message"] ? std::string([cause[@"message"] UTF8String]) : "<unknown>"));
+  error.asObject(runtime).setProperty(runtime, "cause", convertNSDictionaryToJSIObject(runtime, cause));
+  return jsi::JSError(runtime, std::move(error));
+}
+
+/**
+ * Creates JSErrorValue with given stack trace.
+ */
+static jsi::Value createRejectJSErrorValue(jsi::Runtime &runtime, NSDictionary *reason, const std::optional<std::string> &jsInvocationStack)
```

Comment:
Can this be merged with `convertNSDictionaryToJSError` since the last arg is optional anyway?
```suggestion
static jsi::Value createRejectJSErrorValue(jsi::Runtime &runtime, NSDictionary *reason, const std::optional<std::string> &jsInvocationStack = std::nullopt)
```

Reply:
Yes, it's not needed any more.